src/appleseed/renderer/modeling/bssrdf/sss.h

```diff
@@ -288,6 +292,73 @@ void compute_absorption_and_scattering_mfp(
     }
 }
 
+//
+// Functions related to Dwivedi sampling.
+//
+// Reference:
+//
+//   Johannes Meng, Johannes Hanika, Carsten Dachsbacher
+//   Improving the Dwivedi Sampling Scheme,
+//   Journal Computer Graphics Forum Vol. 35 Issue 4, pp. 37-44, July 2016.
+//   https://jo.dreggn.org/home/2016_dwivedi.pdf [1]
+//   https://jo.dreggn.org/home/2016_dwivedi_additional.pdf (supplement material) [2]
+//
+
+// Compute reciprocal of diffusion length for low albedo, [3] Eqn. 10
+static float compute_rcp_diffusion_length_low_albedo(const float albedo)
+{
+    const float a = foundation::rcp(albedo);
+    const float b = exp(-2.0f * a);
+
+    const float x[4] = {
+        1.0f,
+        a * 4.0f - 1.0f,
+        a * (a * 24.0f - 12.0f) + 1.0f,
+        a * (a * (a * 512.0f - 384.0f) + 72.0f) - 3.0f };
+
+    return 1.0f - 2.0f * b * (x[0] + b * (x[1] + b * (x[2] + b * x[3])));
+}
+
+// Compute reciprocal of diffusion length for high albedo, [3] Eqn. 11
+inline float compute_rcp_diffusion_length_high_albedo(const float albedo)
+{
+    const float a = 1.0f - albedo;
+    const float b = sqrt(3.0f * a);
+
+    const float x[5] = {
+        +1.0000000000f,
+        -0.4000000000f,
+        -0.0685714286f,
+        -0.0160000000f,
+        -0.0024638218f };
+
+    return b * (x[0] + a * (x[1] + a * (x[2] + a * (x[3] + a * x[4]))));
+}
+
+// Compute reciprocal of diffusion length.
+// To do this, numerical approximations from [3] are used.
+inline float compute_rcp_diffusion_length(const float albedo)
+{
+    const float a = foundation::clamp(albedo, 0.01f, 0.99f);
+    return a < 0.56f ?
```

Comment:
FYI our formatting convention for this is as follow:

    return a < 0.56f
        ? compute_rcp_diffusion_length_low_albedo(a)
        : compute_rcp_diffusion_length_high_albedo(a);